### PR TITLE
Add count messages per user function

### LINKED ISSUE #257: [Reporting - API] Add method to determine breakdown of messages sent per user on team
**Problem**
Users should be able to see a breakdown of how many messages are sent by users/agents on their team. (This is only relevant for accounts with multiple users/agents on the team.)

**Solution**
Let's create a method in the Reporting context (https://github.com/papercups-io/papercups/blob/master/lib/chat_api/reporting.ex) that counts the number of messages sent per user on the account for a given date range.

Basically this just means counting messages while grouping by `user_id`, and then expanding the `user` object so we can view details like `email`.

The response should look something like this:
```
[
  %{user: %{id: 1, email: "test@test.com"}, count: 28},
  %{user: %{id: 2, email: "example@example.com"}, count: 33},
  # etc
]
```

**Testing**
Please add tests to `test/chat_api/reporting_test.exs` and make sure they pass with `mix test` :) 

**Context**
See https://github.com/papercups-io/papercups/issues/252

**Questions, or need help getting started?**
Feel free to ask below, or [ping us on Slack](https://join.slack.com/t/papercups-io/shared_invite/zt-h0c3fxmd-hZi1Zp8~D61S6GD16aMqmg) :)

(You can also check out our [CONTRIBUTING.md](https://github.com/papercups-io/papercups/blob/master/CONTRIBUTING.md))

### PR DESCRIPTION
**Pull request by estevanjantsk**
### Description

Add method to determine breakdown of messages sent per user on team.
Basically I added `count_messages_per_user` function that returns all messages grouped by the user_id.

### Issue

Resolves #257 

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

File: test/chat_api/reporting_test.exs [hunk 1]
```diff
@@ -42,6 +42,38 @@ defmodule ChatApi.ReportingTest do
...
+
+      message_fixture(account, conversation, %{
+        inserted_at: ~N[2020-09-03 12:00:00],
+        user_id: user_3.id
+      })
+
+      assert [
+               %{count: 2},
+               %{count: 1}
+             ] = Reporting.messages_count_by_date(account.id)
```
Comment (reichert621) [inline test/chat_api/reporting_test.exs:74]: should this be `Reporting.count_messages_per_user`?

File: test/chat_api/reporting_test.exs [hunk 2]
```diff
@@ -42,6 +42,38 @@ defmodule ChatApi.ReportingTest do
              ] = Reporting.messages_by_date(account.id)
     end
 
+    test "count_messages_per_user/1 should return correct number of messages sent per user on team",
+         %{
+           account: account,
+           customer: customer
+         } do
+      account_2 = account_fixture()
+      account_3 = account_fixture()
```
Comment (reichert621) [inline test/chat_api/reporting_test.exs:51]: these extra accounts shouldn't be necessary, we can just add users to the existing `account` from the setup :)

**Review by reichert621 — CHANGES_REQUESTED** [review]
thanks for tackling this as well @estevanjantsk 😁 

the test looks like it won't pass, so just left a couple comments there

File: lib/chat_api/reporting.ex [hunk 3]
```diff
@@ -14,6 +14,15 @@ defmodule ChatApi.Reporting do
     |> Repo.all()
   end
 
+  def count_messages_per_user(account_id) do
```
Comment (reichert621) [inline lib/chat_api/reporting.ex:17]: 🎉 🎉 🎉

File: test/chat_api/reporting_test.exs (hunk 1 above)
Comment (estevanjantsk) [reply test/chat_api/reporting_test.exs:74]: Oh, I changed the function name but I forgot to push the latest commit. 😆 I'm sure this works hahaha. At least on my machine haha.

**Comment by estevanjantsk** [comment]
I will push everything after work. Btw this is a cool project 😊.

File: test/chat_api/reporting_test.exs (hunk 1 above)
Comment (reichert621) [reply test/chat_api/reporting_test.exs:74]: 🎉

**Comment by reichert621** [comment]
thanks @estevanjantsk! 😊

**Comment by estevanjantsk** [comment]
@reichert621 👌 🆗

**Comment by reichert621** [comment]
looks great! 👍

**Review by reichert621 — APPROVED** [review]
thanks for fixing this up!

**Pull request merged**